tests/srw_test.cpp

```diff
@@ -160,6 +213,143 @@ static void timeout_test(session &sess, const std::string &app_name)
 	}
 }
 
+std::string gen_random(const int len) {
+	static const char alphanum[] =
+		"0123456789"
+		"ABCDEFGHIJKLMNOPQRSTUVWXYZ"
+		"abcdefghijklmnopqrstuvwxyz";
+
+	std::string result;
+	result.reserve (len);
+
+	for (int i = 0; i < len; ++i) {
+			result += alphanum[rand() % (sizeof(alphanum) - 1)];
+	}
+	return result;
+}
+
+template<class T>
+T pack_unpack(const T &v)
+{
+	msgpack::sbuffer buf;
+	msgpack::pack(buf, v);
+
+	msgpack::unpacked msg;
+	msgpack::unpack(&msg, buf.data(), buf.size());
+
+	return msg.get().as<T>();
+}
+
+static void localnode_data_serialization_test()
+{
+	{
+		dnet_id a = { "0123455678", 2, 3 };
+		dnet_id b = pack_unpack(a);
+		BOOST_REQUIRE_EQUAL(a.group_id, b.group_id);
+		BOOST_REQUIRE_EQUAL(std::string((const char *)a.id), std::string((const char *)b.id));
+	}
+	{
+		dnet_addr a = { "abc", 5, 8 };
+		dnet_addr b = pack_unpack(a);
+		BOOST_REQUIRE_EQUAL(a.addr_len, b.addr_len);
+		BOOST_REQUIRE_EQUAL(std::string((const char *)a.addr, a.addr_len), std::string((const char *)b.addr, b.addr_len));
+		BOOST_REQUIRE_EQUAL(a.family, b.family);
+	}
+	{
+		dnet_time a = { 5, 8 };
+		dnet_time b = pack_unpack(a);
+		BOOST_REQUIRE_EQUAL(a.tsec, b.tsec);
+		BOOST_REQUIRE_EQUAL(a.tnsec, b.tnsec);
+	}
+	{
+		dnet_file_info a = { 3, "abc", 5, 8, 10, { 2, 3} };
+		dnet_file_info b = pack_unpack(a);
+#define CMP(x) BOOST_REQUIRE_EQUAL(a.x, b.x)
+		CMP(flen);
+		CMP(record_flags);
+		CMP(size);
+		CMP(offset);
+		CMP(mtime.tsec);
+		CMP(mtime.tnsec);
+#undef CMP
+		BOOST_REQUIRE_EQUAL(
+			std::string((const char *)a.checksum, sizeof(a.checksum)),
+			std::string((const char *)b.checksum, sizeof(b.checksum))
+		);
+	}
+	{
+		dnet_async_service_result a = {
+			{ "abc", 5, 8 },
+			{ 3, "abc", 5, 8, 10, { 2, 3} },
+			"file path"
+		};
+		dnet_async_service_result b = pack_unpack(a);
+		BOOST_REQUIRE_EQUAL(a, b);
+	}
+}
+
+static void localnode_test(session &sess, int group_id)
+{
+	using namespace cocaine::framework;
+
+	service_manager_t::endpoint_t endpoint("127.0.0.1", global_data->locator_port);
+	auto manager = service_manager_t::create(endpoint);
+
+	auto localnode = manager->get_service<cocaine::framework::direct_access_service_t>("localnode");
+
+	key key(gen_random(8));
+	key.transform(sess);
+
+	key.set_group_id(group_id);
+
+	auto value = gen_random(15);
+
+	dnet_async_service_result write_result;
+	dnet_async_service_result lookup_result;
+	{
+		auto &result = write_result;
+		auto deferred = localnode->write_data(key.id(), value, 0);
+		BOOST_REQUIRE_NO_THROW(result = deferred.next());
+		BOOST_REQUIRE_EQUAL(deferred.valid(), true);
+		BOOST_CHECK_GT(result.file_info.size, 0);
```

Comment:
Because `write` and `lookup` commands return different sizes -- `lookup` returns user data size while `write` return user data size plus some extra (some header size, may be). Or vice versa.